### PR TITLE
26 config.toml

### PR DESCRIPTION
**Pull request by ndaultryball1**
- [x] Write some tests. make new test in tests/skeleton.rs to check config is copied over.

### CODE REVIEW

File: src/skeleton.rs [hunk 1]
```diff
@@ -87,6 +88,34 @@ impl Skeleton {
...
+        let walker = GlobWalkerBuilder::new(&base_path, "/**/.cargo/config.toml")
+            .build()
+            .context("Failed to scan the files in the current directory.")?;
+        let mut configs = vec![];
+        for config in walker {
+            match dbg!(config) {
+                Ok(manifest) => {
+                    let absolute_path = manifest.path().to_path_buf();
+                    let contents = fs::read_to_string(&absolute_path)?;
+                    // dbg!(absolute_path);
```
Comment (ndaultryball1) [inline src/skeleton.rs:100]: Remove debug statements

File: src/skeleton.rs [hunk 2]
```diff
@@ -27,7 +28,7 @@ impl Skeleton {
             .context("Failed to scan the files in the current directory.")?;
         let mut manifests = vec![];
         for manifest in walker {
-            match manifest {
+            match dbg!(manifest) {
```
Comment (ndaultryball1) [inline src/skeleton.rs:31]: remove debug

File: src/skeleton.rs [hunk 3]
```diff
@@ -87,6 +88,34 @@ impl Skeleton {
...
+                                &absolute_path
+                            )
+                        })?;
+                    configs.push(Manifest {
+                        relative_path,
+                        contents,
+                    });
+                }
+
+                Err(_) => todo!(),
```
Comment (ndaultryball1) [inline src/skeleton.rs:116]: Handle this error

File: src/skeleton.rs [hunk 4]
```diff
@@ -87,6 +88,34 @@ impl Skeleton {
...
+                Ok(manifest) => {
+                    let absolute_path = manifest.path().to_path_buf();
+                    let contents = fs::read_to_string(&absolute_path)?;
+                    // dbg!(absolute_path);
+                    // dbg!(contents);
+
+                    let relative_path = pathdiff::diff_paths(&absolute_path, &base_path)
+                        .ok_or_else(|| {
+                            anyhow::anyhow!(
+                                "Failed to compute relative path of manifest {:?}",
```
Comment (ndaultryball1) [inline src/skeleton.rs:106]: Make this config specific message

File: src/skeleton.rs [hunk 5]
```diff
@@ -87,6 +88,34 @@ impl Skeleton {
...
             }
         }
+        let walker = GlobWalkerBuilder::new(&base_path, "/**/.cargo/config.toml")
+            .build()
+            .context("Failed to scan the files in the current directory.")?;
+        let mut configs = vec![];
+        for config in walker {
+            match dbg!(config) {
+                Ok(manifest) => {
+                    let absolute_path = manifest.path().to_path_buf();
```
Comment (ndaultryball1) [inline src/skeleton.rs:98]: Should we de-duplicate this with the manifest case?

File: src/skeleton.rs [hunk 6]
```diff
@@ -9,6 +9,7 @@ use std::str::FromStr;
 #[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq)]
 pub struct Skeleton {
     pub manifests: Vec<Manifest>,
+    pub configs: Vec<Manifest>,
```
Comment (ndaultryball1) [inline src/skeleton.rs:12]: Should we have a "raw file" struct with a manifest wrapper, or is this abuse of manifest ok?

File: src/skeleton.rs [hunk 7]
```diff
@@ -214,6 +244,17 @@ impl Skeleton {
                 }
             }
         }
+        
+        for config in &self.configs {
+
```
Comment (ndaultryball1) [inline src/skeleton.rs:249]: Run cargo fmt

File: src/skeleton.rs [hunk 8]
```diff
@@ -214,6 +244,17 @@ impl Skeleton {
...
+            let config_file_path = base_path.join(&config.relative_path);
```
Comment (ndaultryball1) [inline src/skeleton.rs:250]: Does this need to be de-duplicated too?

File: src/skeleton.rs [hunk 9]
```diff
@@ -87,6 +88,34 @@ impl Skeleton {
                 },
             }
         }
+        let walker = GlobWalkerBuilder::new(&base_path, "/**/.cargo/config.toml")
```
Comment (ndaultryball1) [inline src/skeleton.rs:91]: We can just look in the top level rather than walking

**Review by alsuren — APPROVED** [review]

**Review by Marmiz — APPROVED** [review]

**Review by LukeMathWalker — APPROVED** [review]

**Pull request merged**